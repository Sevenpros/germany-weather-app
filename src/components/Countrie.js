const Countrie = () => (
  <div className="container">
    <div className="headline" />
    <div className="grid-container">
      <h1>Country</h1>
      <h2>Confirmed Cases</h2>
    </div>
  </div>
);

export const Detail = () => (
  <ul>
    <li className="c-details">Country: Germany</li>
    <li className="c-details">Deaths: 100</li>
    <li className="c-details">New Case: 40</li>
    <li className="c-details">Recover: 67</li>
  </ul>
);

export default Countrie;
