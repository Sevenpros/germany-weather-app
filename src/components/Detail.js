const Detail = () => (
  <ul>
    <li className="c-details">Country: Germany</li>
    <li className="c-details">Deaths: 100</li>
    <li className="c-details">New Case: 40</li>
    <li className="c-details">Recover: 67</li>
  </ul>
);

export default Detail;
