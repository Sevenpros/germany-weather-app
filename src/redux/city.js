
import { createAsyncThunk } from '@reduxjs/toolkit';
import cityList from '../components/citylist';

const initialState = {
  cities: [],
};

const LOAD_CITIES = 'LOAD_CITIES';

const cityReducer = (state = initialState, action) => {
  switch (action.type) {
    case `${LOAD_CITIES}/fulfilled`:
      return { ...state, cities: action.payload };
    default: return state;
  }
};

const getCityInfo = async (city) => {
  const { lat, lon } = city;
  const key = 'c63bf5c12f2f784cd947f5cc3a11296e';
  const api = `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${key}`;
  const response = await (await fetch(api)).json();
  return response;
};
const toCelcius = (kelvin) => Math.round((kelvin - 273.15));

const transformData = (city) => {
  const { lat, lon } = city.coord;
  const { temp, feels_like } = city.main;
  const [currentWeather] = city.weather;
  const { description } = currentWeather;
  const celcius = toCelcius(temp);
  const feels = toCelcius(feels_like);
  const obj = {};
  obj.id = city.id;
  obj.temp = celcius;
  obj.feels_like = feels;
  obj.lat = lat;
  obj.lon = lon;
  obj.name = city.name;
  obj.description = description;
  return obj;
};

export const fetchCities = createAsyncThunk(LOAD_CITIES, async () => {
  const cities = [];
  cityList.filter((city) => {
    getCityInfo(city).then((res) => {
      cities.push(transformData(res));
    });
  });
  return cities;
});
export default cityReducer;
