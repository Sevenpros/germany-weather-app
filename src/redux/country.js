import { createAsyncThunk } from '@reduxjs/toolkit';

const initialState = {
  countries: [],
  details: {},
};

const LOAD_COUNTRIES = 'LOAD_COUNTRIES';
const LOAD_DETAILS = 'LOAD_DETAILS';
const API_URL = 'https://api.covid19api.com/summary';

const countryReducer = (state = initialState, action) => {
  switch (action.type) {
    case `${LOAD_COUNTRIES}/fulfilled`:
      return { ...state, countries: action.payload };
    case LOAD_DETAILS: {
      const details = state.countries.find((country) => (country.ID === action.id));
      return { ...state, details }; }
    default: return state;
  }
};

export const fetchCountries = createAsyncThunk(LOAD_COUNTRIES, async () => {
  const Response = await fetch(API_URL);
  const result = await Response.json();
  return result.Countries;
});

export const loadDetails = (id) => ({ type: LOAD_DETAILS, id });
export default countryReducer;
