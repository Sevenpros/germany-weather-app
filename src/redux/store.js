import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import countryReducer from './country';

const store = configureStore({
  reducer: {
    countryReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});
export default store;
