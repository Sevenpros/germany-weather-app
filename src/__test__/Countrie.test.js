import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Countrie from '../components/Countrie';
import Detail from '../components/Detail';
import store from '../redux/store';
import { loadDetails } from '../redux/country';

describe('Countries tests', () => {
  it('render Countries test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Countrie />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('render Details test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Detail />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should return an action with type LOAD_DETAILS', () => {
    const id = 1;
    const action = loadDetails(id);
    expect(action.type).toBe('LOAD_DETAILS');
  });
  it('should return an action with expected ID', () => {
    const id = 1;
    const action = loadDetails(id);
    expect(action.id).toBe(1);
  });
});
