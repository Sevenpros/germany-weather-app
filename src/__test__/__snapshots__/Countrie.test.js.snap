// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries tests render Countries test 1`] = `
<div
  className="container"
>
  <div
    className="headline"
  />
  <div
    className="grid-container"
  >
    <h1>
      Country
    </h1>
    <h2>
      Confirmed Cases
    </h2>
  </div>
</div>
`;

exports[`Countries tests render Details test 1`] = `
<ul>
  <li
    className="c-details"
  >
    Country: Germany
  </li>
  <li
    className="c-details"
  >
    Deaths: 100
  </li>
  <li
    className="c-details"
  >
    New Case: 40
  </li>
  <li
    className="c-details"
  >
    Recover: 67
  </li>
</ul>
`;
